agent/api/task/task_test.go

```diff
@@ -4983,6 +4983,35 @@ func TestToHostResources(t *testing.T) {
 		},
 	}
 
+	// A combination of containers with different configs with memory sourcing from different sources
+	testTask5 := &Task{
```

Comment:
This test may not be able to catch the gap that we are fixing, since the test uses a single hostConfig for all containers, In that case, the existing implementation would've worked fine. We should create at least two different hostConfigs with different memoryReservation, such that the existing implementation would fail

Reply:
Ack - additionally, existing implementation fails due to payload Agent receives does not have all the HostConfig fields. So the unit test and it's hostConfig need to be defined in such a way as well to actually repro the case. Pushed changes